worker/src/main/java/gov/cms/ab2d/worker/processor/PatientClaimsProcessorImpl.java

```diff
@@ -104,7 +107,24 @@ private void handleException(StreamHelper helper, String data, Exception e) thro
         var patient = request.getPatientDTO();
         var attTime = request.getAttTime();
 
-        Bundle eobBundle = bfdClient.requestEOBFromServer(patient.getPatientId(), request.getSinceTime());
+        OffsetDateTime start = OffsetDateTime.now();
+        Bundle eobBundle;
+        try {
+            eobBundle = bfdClient.requestEOBFromServer(patient.getPatientId(), request.getSinceTime());
+            logManager.log(LogManager.LogType.KINESIS,
+                    new BeneficiarySearchEvent(request.getUser(), request.getJob(), request.getContractNum(),
+                            start, OffsetDateTime.now(),
+                            request.getPatientDTO() != null ? request.getPatientDTO().getPatientId() : null,
+                            "SUCCESS"));
```

Comment:
We'll need to be prepared to refactor Kinesis logger to be asynchronous (not in this PR though). Reason: we are making hundreds of thousands (if not millions) EOB calls to BFD to fulfill a single export, which in total is accounting for most of our job processing time. Kinesis logger now makes additional _blocking_ network calls to AWS, which are going to contribute to the overall job processing time given the fact there are also hundreds of thousands of those.

Of course we won't attempt a refactor until we confirm in New Relic that the logger adds a significant overhead, but we should keep this in the back of our minds.

Reply:
Good point.